File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/models/Restaurant.java
package models;

import java.util.List;

public class Restaurant {
    private String id;
    private String name;
    private String address;
    private List<FoodItem> menu;

    public Restaurant(String id, String name, String address, List<FoodItem> menu) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.menu = menu;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getAddress() {
        return address;
    }

    public List<FoodItem> getMenu() {
        return menu;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/models/src/Restaurant.cpp
#include "../include/Restaurant.h"

Restaurant::Restaurant(const std::string& id, const std::string& name,
                       const std::string& address, const std::vector<FoodItem>& menu)
    : id(id), name(name), address(address), menu(menu) {}

const std::string& Restaurant::getId() const {
    return id;
}

const std::string& Restaurant::getName() const {
    return name;
}

const std::string& Restaurant::getAddress() const {
    return address;
}

const std::vector<FoodItem>& Restaurant::getMenu() const {
    return menu;
}
